packages/block-library/src/query/edit/query-placeholder.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import {
+	useBlockProps,
+	__experimentalBlockVariationPicker,
+} from '@wordpress/block-editor';
+import { createBlocksFromInnerBlocksTemplate } from '@wordpress/blocks';
+
+const QueryPlaceholder = ( { clientId, name, setAttributes } ) => {
+	const { blockType, defaultVariation, variations } = useSelect(
+		( select ) => {
+			const {
+				getBlockVariations,
+				getBlockType,
+				getDefaultBlockVariation,
+			} = select( 'core/blocks' );
+
+			return {
+				blockType: getBlockType( name ),
+				defaultVariation: getDefaultBlockVariation( name, 'block' ),
+				variations: getBlockVariations( name, 'block' ),
+			};
+		},
+		[ name ]
+	);
+	const { replaceInnerBlocks } = useDispatch( 'core/block-editor' );
+	const blockProps = useBlockProps();
+	return (
+		<div { ...blockProps }>
+			<__experimentalBlockVariationPicker
+				icon={ blockType?.icon?.src }
+				label={ blockType?.title }
+				variations={ variations }
+				onSelect={ ( nextVariation = defaultVariation ) => {
+					if ( nextVariation.attributes ) {
+						setAttributes( nextVariation.attributes );
+					}
+					if ( nextVariation.innerBlocks ) {
+						replaceInnerBlocks(
+							clientId,
+							createBlocksFromInnerBlocksTemplate(
+								nextVariation.innerBlocks
+							),
+							false
+						);
+					}
+				} }
```

Comment:
Columns block has the option to skip selection and go with the default setup. Should we add it here as well? See:

https://github.com/WordPress/gutenberg/blob/263c5ace7e7cb4b79c80dae18e58c6e5dc788a71/packages/block-library/src/columns/edit.js#L242

Reply:
I'm not sure what problem the "Skip" option solves. Forcing a selection of one of the options seems perfectly reasonable to me.

Reply:
I also think that skip is not necessary in Query, for now at least, as it's more hustle to add/remove blocks from Query later, than just selecting a variation that might suit the user more.